apps/namadillo/src/atoms/balance/services.ts

```diff
@@ -39,7 +41,15 @@ export function shieldedSync(
   const emitter = new EventEmitter<ShieldedSyncEventMap>();
 
   worker.onmessage = (event: MessageEvent<Events>) => {
-    emitter.emit(event.data.type, event.data);
+    if (event.data.type === SdkEvents.ProgressBarStarted) {
+      emitter.emit(event.data.type, event.data);
+    }
+    if (event.data.type === SdkEvents.ProgressBarIncremented) {
+      emitter.emit(event.data.type, event.data);
+    }
+    if (event.data.type === SdkEvents.ProgressBarFinished) {
+      emitter.emit(event.data.type, event.data);
+    }
```

Comment:
Those ifs are a bit ugly but I don't think we can make it better and preserver correct types 😅